### PR TITLE
3.x - implemented Router::reverseToArray()

### PR DESCRIPTION
**Pull request by bancer**
Resolves https://github.com/cakephp/cakephp/issues/10092

### CODE REVIEW

File: tests/TestCase/Routing/RouterTest.php [hunk 1]
```diff
@@ -2753,6 +2767,53 @@ public function testReverseWithExtension()
...
+        ];
+        $actual = Router::reverseToArray($params);
+        $expected = [
+            'lang' => 'eng',
+            'controller' => 'posts',
+            'action' => 'view',
+            123,
+            '?' => ['foo' => 'bar'],
+        ];
+        $this->assertEquals($expected, $actual);
```
Comment (markstory) [inline tests/TestCase/Routing/RouterTest.php]: Is this test doing much?

Comment (bancer) [inline tests/TestCase/Routing/RouterTest.php]: Well, I used the params from `testReverseArrayQuery()`. Any suggestions?

File: tests/TestCase/Routing/RouterTest.php [hunk 2]
```diff
@@ -2753,6 +2767,55 @@ public function testReverseWithExtension()
...
+    {
+        Router::connect('/:lang/:controller/:action/*', [], ['lang' => '[a-z]{3}']);
+        $params = [
+            'lang' => 'eng',
+            'controller' => 'posts',
+            'action' => 'view',
+            'pass' => [123],
+            'url' => ['url' => 'eng/posts/view/123', 'foo' => 'bar', 'baz' => 'quu'],
+            'paging' => [],
+            'models' => [],
```
Comment (markstory) [inline tests/TestCase/Routing/RouterTest.php]: This test is better. It better captures which keys are omitted and which are preserved.

**Comment by markstory** [comment]
Related to #10113

**Pull request merged**